pkg/scheduler/scheduling_application.go

```diff
@@ -163,6 +167,7 @@ func (sa *SchedulingApplication) removeAllocationAsk(allocKey string) int {
 		deltaPendingResource = sa.pending
 		sa.pending = resources.NewResource()
 		sa.requests = make(map[string]*schedulingAllocationAsk)
+		sa.sortedRequestsDirty = true
```

Comment:
Why does the array turn to dirty when we remove an ask? I believe removing an element from a sorted list does not make the list unsorted.

Reply:
`sortedRequest` is a copied sorted slice and not the original map. The value is removed from the original map but we need to recreate the sorted slice before we use it again otherwise we might have an entry in the slice that does not exist causing scheduling errors.